### DISCUSSION TITLE
All definitions in Jolt.cmake are PUBLIC

### DISCUSSION THREAD

**Discussion by Please-just-dont** [jrouwe/JoltPhysics#1307] [Q&A]
Which means any user of the library, at least through CMake, will inherit those definitions.  Why aren't they PRIVATE?

**Comment by Please-just-dont** [comment]
Hmm, I guess they're all prefixed with JPH_, the only ones that aren't are _DEBUG, NDEBUG, and _HAS_EXCEPTIONS=0. I don't have a conflict or anything I'm just wondering why.

**Comment by jrouwe** [comment]
They aren't PRIVATE since most of them affect generated code for the application. E.g. if `JPH_DEBUG_RENDERER` is defined for the library but not for the application, then you will get a linker error because `Shape::GetSubmergedVolume` has a different number of parameters based on if `JPH_DEBUG_RENDERER` is defined or not. So effectively the library defines another function than what the application is looking for. 

Most of the defines cause the code to be different enough to either cause linker errors or crashes since class sizes / virtual function pointer tables are incompatible.
